test/hooks_test.dart

```diff
@@ -666,6 +666,111 @@ main() {
         });
       });
     });
+
+    group('useImperativeHandle -', () {
+      var mountNode = new DivElement();
+      ReactDartFunctionComponentFactoryProxy UseImperativeHandleTest;
+      ButtonElement incrementButton;
+      ButtonElement reRenderButtonRef1;
+      ButtonElement reRenderButtonRef2;
+      Ref noDepsRef;
+      Ref emptyDepsRef;
+      Ref depsRef;
+      StateHook<int> count;
+
+      setUpAll(() {
+        var NoDepsComponent = react.forwardRef((props, ref) {
+          count = useState(0);
+
+          useImperativeHandle(
+            ref,
+            () => ({
+              'increment': () {
+                count.setWithUpdater((prev) => prev + 1);
+              }
+            }),
+          );
+
+          return react.div({'ref': ref}, count.value);
+        });
+
+        var EmptyDepsComponent = react.forwardRef((props, ref) {
+          var count = useState(0);
+
+          useImperativeHandle(ref, () => count.value, []);
+
+          return react.Fragment({}, [
+            react.div({'ref': ref}, count.value),
+            react.button({
+              'ref': (ref) => reRenderButtonRef1 = ref,
+              'onClick': (_) => count.setWithUpdater((prev) => prev + 1),
+            }, []),
+          ]);
+        });
+
+        var DepsComponent = react.forwardRef((props, ref) {
+          var count = useState(0);
+
+          useImperativeHandle(ref, () => count.value, [count.value]);
+
+          return react.Fragment({}, [
+            react.div({'ref': ref}, count.value),
+            react.button({
+              'ref': (ref) => reRenderButtonRef2 = ref,
+              'onClick': (_) => count.setWithUpdater((prev) => prev + 1),
+            }, []),
+          ]);
+        });
+
+        UseImperativeHandleTest = react.registerFunctionComponent((Map props) {
+          noDepsRef = useRef();
+          emptyDepsRef = useRef();
+          depsRef = useRef();
+
+          return react.Fragment({}, [
+            NoDepsComponent({'ref': noDepsRef}, []),
+            react.button({
+              'ref': (ref) => incrementButton = ref,
+              'onClick': (_) => noDepsRef.current['increment'](),
+            }, [
+              '+'
+            ]),
+            EmptyDepsComponent({'ref': emptyDepsRef}),
+            DepsComponent({'ref': depsRef}),
+          ]);
+        });
+
+        react_dom.render(UseImperativeHandleTest({}), mountNode);
+      });
```

Comment:
#nit I'd add a line here, personally 
```suggestion
      });

```